examples/viewports.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * viewports.c
+ * Copyright (C) 2023 Igalia S.L.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include "../core/cog.h"
+
+typedef struct {
+    CogViewport *views;
+    gsize        current_index;
+} TimeoutData;
+
+static gboolean
+on_timeout_tick(TimeoutData *data)
+{
+    if (++data->current_index >= cog_viewport_get_n_views(data->views))
+        data->current_index = 0;
+
+    CogView *view = cog_viewport_get_nth_view(data->views, data->current_index);
+    g_message("Set visible view %zu <%p>", data->current_index, view);
+    cog_viewport_set_visible_view(data->views, view);
+
+    return G_SOURCE_CONTINUE;
+}
+
+int
+main(int argc, char *argv[])
+{
+    if (argc < 2) {
+        g_printerr("Usage: %s <URL> [URL...]\n", argv[0]);
+        return EXIT_FAILURE;
+    }
+
+    g_set_prgname("view-stacks");
+
+    g_autoptr(CogShell) shell = cog_shell_new(g_get_prgname(), FALSE);
+    g_autoptr(GError)   error = NULL;
+
+    g_autoptr(CogPlatform) platform = cog_platform_get();
+    if (!cog_platform_setup(platform, shell, g_getenv("COG_PLATFORM_PARAMS") ?: "", &error))
+        g_error("Cannot configure platform: %s", error->message);
+
+    g_autoptr(GMainLoop) loop = g_main_loop_new(NULL, FALSE);
+
+    CogViewport *viewport0 = cog_shell_get_viewport(shell);
+
+    for (int i = 1; i < argc; i++) {
+        g_autoptr(CogView) view = cog_view_new(NULL);
+        cog_platform_init_web_view(platform, WEBKIT_WEB_VIEW(view));
```

Comment:
Having to call `cog_platform_init_web_view()` by hand is annoying. I'm sure we can change it in the future, we can add a `CogView:viewport` property (most of the code is already there) and then view code that needs late initialization when the view is known to have a viewport assigned can monitor the property. Currently what the `x11`, `drm`, and `gtk4` plug-ins do is that (late initializations) or other configurations which could be moved to the `GObject.constructed` virtual function. 

Anyway, this is not something we need to change now, I am writing my thoughts here so we can add an issue and make a patch later.

Reply:
Issue created to handle this later: https://github.com/Igalia/cog/issues/678 🏗️